### DISCUSSION TITLE
Consider switching the compiler to a (weak) copyleft license?

### DISCUSSION THREAD

**Discussion by jeanas** [typst/typst#5018] [General]
Hi there,

Recently I mentioned Typst to a mathematician as an alternative to *TeX, and his immediate reaction from opening the Typst website was: Is this yet another piece of software that starts as open source to embrace the world so that it can make more money afterwards by going closed source? (Thinking of examples like Elastic, MongoDB, Redis, RHEL, …)

I believe the Typst developers when they say the compiler will always remain open source. On the other hand, given how much academia suffers from rapacious publishers which take a toll on the scientific output that is disproportionate to the value they add, I can understand the fear of getting locked into proprietary software.

For this reason, I'm wondering if you would consider switching the Typst compiler to a copyleft license. This doesn't need to be a strong copyleft license like the GPL (all code that incorporates the licensed code must be under the same license), but can be a weak copyleft license like MPL or LGPL (modifications of the code must be released under the same license, but it can be used in proprietary code). Combined with the blending of copyrights from the main developers and other contributors (no CLA), this would provide a strong, reassuring guarantee that the compiler will remain open source going forward.

Thanks!

**Comment by Enter-tainer** [comment]
> this would provide a strong, reassuring guarantee that the compiler will remain open source going forward.

i dont understand why this make sense. the code owner can always close source whenever they like. this is unrelated to license.

**Reply by Enter-tainer** [reply]
doing license tricks doesnt make a lot of sense to me. the key to solve your concern is that typst gmbh need to develop a healthy business model. so it can make (a lot of) money without sacrificing end user experiences.

**Reply by jeanas** [reply]
Unless there is a copyright assignment process (CLA), the copyright is jointly held by all authors, which in the case of Typst also includes external contributors (although they account for a minority of contributions), so you cannot relicense the code without agreement from all of them.

What you call a "license trick" is actually pretty standard (see: Linux kernel, LibreOffice, MediaWiki, GTK, etc. etc.).

**Reply by Enter-tainer** [reply]
developing complex software like typst requires extensive human labor. this is nearly impossible without skilled fulltime engineers.

if they cannot make money from that to support their life, they will either abandon the repo(note that they may earn a lot if they join google or something), or try to re-license it to make more money.

changing the license to something more copyleft, still doesnt solve the core problem.

**Reply by Enter-tainer** [reply]
i'm ok if the repo owner of typst decides to switch to whatever copyleft license. but i dont think it would help much and solve your concern.

if what you imagine really happens, switching to copyleft license will only gives us an abandoned repo.
